src/connector/src/parser/mod.rs

```diff
@@ -310,32 +312,64 @@ impl SourceStreamChunkRowWriter<'_> {
-                    Ok(output) => Ok(output),
-
-                    // Throw error for failed access to primary key columns.
-                    Err(e) if desc.is_pk => Err(e),
-                    // Ignore error for other columns and fill in `NULL` instead.
-                    Err(error) => {
-                        // TODO: figure out a way to fill in not-null default value if user specifies one
-                        // TODO: decide whether the error should not be ignored (e.g., even not a valid Debezium message)
-                        // TODO: not using tracing span to provide `split_id` and `offset` due to performance concern,
-                        //       see #13105
-                        tracing::warn!(
-                            %error,
-                            split_id = self.row_meta.as_ref().map(|m| m.split_id),
-                            offset = self.row_meta.as_ref().map(|m| m.offset),
-                            column = desc.name,
-                            "failed to parse non-pk column, padding with `NULL`"
-                        );
-                        Ok(A::output_for(Datum::None))
+            match (&desc.column_type, &desc.additional_column_type) {
+                (&SourceColumnType::Offset | &SourceColumnType::RowId, _) => {
+                    // SourceColumnType is for CDC source only.
+                    Ok(A::output_for(
+                        self.row_meta
+                            .as_ref()
+                            .and_then(|row_meta| row_meta.value_for_column(desc))
+                            .unwrap(), // handled all match cases in internal match, unwrap is safe
+                    ))
+                }
+                (&SourceColumnType::Meta, _)
+                    if matches!(
+                        &self.row_meta.map(|ele| ele.meta),
+                        &Some(SourceMeta::Kafka(_) | SourceMeta::DebeziumCdc(_))
+                    ) =>
+                {
+                    // SourceColumnType is for CDC source only.
+                    return Ok(A::output_for(
+                        self.row_meta
+                            .as_ref()
+                            .and_then(|row_meta| row_meta.value_for_column(desc))
+                            .unwrap(), // handled all match cases in internal match, unwrap is safe
+                    ));
+                }
+                (
+                    _,
+                    &AdditionalColumnType::Timestamp
+                    | &AdditionalColumnType::Partition
+                    | &AdditionalColumnType::Filename
+                    | &AdditionalColumnType::Offset
+                    | &AdditionalColumnType::Header,
+                    // AdditionalColumnType::Unspecified is means it comes from message payload
+                    // AdditionalColumnType::Key is processed in normal process, together with Unspecified ones
+                ) => {
+                    todo!()
```

Comment:
Ignoring is better?

Reply:
we are handling additional columns other than key here

Reply:
So what is better? `panic` or returning `Null`? Anyway, here should not be a `todo!()`, I think.